File: Algorithms/editDistance.java
public class Solution {
    public int minDistance(String word1, String word2) {
        int[][] D = new int[word1.length()+1][word2.length()+1];
        for (int i=0; i <= word1.length(); i++)
            D[i][0] = i;
        for (int i=0; i <= word2.length(); i++)
            D[0][i] = i;
        for (int i=1; i <= word1.length(); i++) {
            for (int j=1; j <= word2.length(); j++) {
                D[i][j] = Math.min(D[i][j-1], D[i-1][j]) + 1;
                D[i][j] = Math.min(D[i][j], D[i-1][j-1] + (word1.charAt(i-1) == word2.charAt(j-1)? 0:1));
            }
        }
        return D[word1.length()][word2.length()];
    }
}

File: Algorithms/editDistance.py
class Solution:
    # @return an integer
    def minDistance(self, word1, word2):
        distance = [[0 for x in range(len(word2) + 1)] for x in range(len(word1) + 1)] 
        for i in range(len(word1) + 1):
            distance[i][0] = i
        for i in range(len(word2) + 1):
            distance[0][i] = i
        for i in range(1, len(word1)+1):
            for j in range(1, len(word2)+1):
                distance[i][j] = min(distance[i][j-1], distance[i-1][j]) + 1
                distance[i][j] = min(distance[i][j], 
                        distance[i-1][j-1] + (0 if word1[i-1] == word2[j-1] else 1))
        return distance[len(word1)][len(word2)]

